packages/core/lib/runner.js

```diff
@@ -493,6 +493,12 @@ function handleScriptHook(hook, script, hookEvents, contextVars = {}) {
 
     const name = script[hook].engine || 'http';
     const engine = engines.find((e) => e.__name === name);
+
```

Comment:
similar logic to https://github.com/artilleryio/artillery/pull/2101 for consistency